lib/KeyWrapper.hpp

```diff
@@ -70,5 +71,20 @@ namespace LibConfig
       *this = other;
     }
 
+    KeyWrapper(const std::string &dataStr, KeyType t)
```

Comment:
I would use the initialiser-list syntax here. The `memcpy` would stay in the body, but te reset could be replaced by a call to the constructor and both other fields are trivially initialisable.

Reply:
As you wish, but KeyWrapper was made to hide memcpy and memory allocation, so I don't really understand.

Reply:
Well yeah, the memcpy would still be in the KeyWrapper, we're just avoiding initialising things twice. Something like:
```cpp
    KeyWrapper(const std::string &dataStr, KeyType t)
    : data(new char[dataStr.length()])
    , size(dataStr.length())
    , type(t)
    {
      std::memcpy(data.get(), dataStr.c_str(), size);
    }
```